### DISCUSSION TITLE
Bindings for mute / push to talk.

### DISCUSSION THREAD

**Discussion by PeachiPrism** [stoatchat/discussions#278] [Feature Suggestions]
Would like to be able to have a bind that toggles mute on voice channels that I can set.

I currently use the Pause Break button to mute on discord and this has been of great use to me as im able to mute if someone comes in the room without having the app open (mostly when im in a game).
I don't even mind missing out on some of the lesser used discord binds, Toggle Mute and Push To Talk are good enough.

**Comment by egditch** [comment]
User keybinds to mute toggle/push to talk and stuff like that is the only thing that I feel like is missing from revolt and holds me back from using it full-time.

**Comment by suspiciouslyBee** [comment]
necromancing this post as these functions are still essential towards a convenient voice chat. in some software i even bind a push-to-*mute* to let me chat somewhere like in a game. please keep this in mind

**Reply by jhebert8** [reply]
I agree with the previous post, definitely push-to-talk is necessary to fully replace Discord.

**Comment by Servonius** [comment]
+1 for toggle mute keybinding, been using it for years and can no longer live without it ;(

**Comment by PeachiPrism** [comment]
I am surprised that this is still an issue several years on. Not quite sure why people are looking at Revolt/Stoat as an alternative given recent events when the fundamentals are the same as Discord. If you care about your data privacy and not _ever_ wanting to have to give your facial ID up you should be looking at something decentralised like Matrix instead.

No hate to the developers though like I get it's been a side project and it's not easy to 1 to 1 feature copy of Discord but if you are going to swap platforms and take some sacrifices in functionality that have been outstanding for years you might as well go all the way and actually own your data.

**Comment by RoBOticRebel** [comment]
Push to talk is quite essential imo

**Reply by Cookie0fPower** [reply]
Agreed

**Comment by steamypickle** [comment]
I know this was from Dec 2021 but can we upvote this harder please.

**Comment by arceusplayer11** [comment]
Yes, please add this! I like having control over what goes through my mic and what doesn't; either when I'm playing games and don't want the things I mutter to come through, or if someone wants to talk to me IRL while I'm on a voice call; I prefer holding a button whenever I want to speak. Could an implementation also come with A: a sound effect when push to talk is activated, and B: the ability to use gamepad ontroller buttons as push to talk keys as well?

**Comment by nayebynny** [comment]
As much as i would like to migrate from discord to here I can't be on a vc without push to talk, it's extremely essential thing for me. I really hope it'll get added soon.

**Reply by Cookie0fPower** [reply]
Agreed

**Comment by piectomek** [comment]
Without it I'm doomed. I need "press to mute"  keybind sometimes when holding button. Please add it 🙏

**Comment by TheDarkGodOMG** [comment]
+1 for definable hotkeys needed for mute/deafen.  Essential for me to be able to use this as a daily driver software.

To give the specific reason: I stream on Twitch/YouTube/etc. so I need to be able to mute my team voice call but still leave my microphone output unmuted to talk to chat.  I currently have Discord hotkeyed so I can mute my team conversation with a single press and continue to interact with chat, then unmute to communicate with my team when necessary.

Without the hotkey, my only option is to mute the microphone completely, which simply does not fit the use case.

**Comment by sdaugherty** [comment]
+1 - even with perfect noise suppression and voice activation that's so good it borders on omniscience, push to talk is critical as both a privacy feature and as a courtesy to fellow chatters.

Hot mics ruin lives and eardrums.

**Comment by ashtarah** [comment]
Lack of push-to-talk options is the only thing keeping my guild from switching to Stoat.

**Reply by jhebert8** [reply]
Same

**Comment by aredon** [comment]
This is a pretty major feature. +1

**Comment by MrWilza** [comment]
Yep, lack of PTT is a deal-breaker for me and many other, I was kind of shocked to learn that such a basic feature is missing

**Comment by Mardraced** [comment]
This is a basic feature of any voice chat. Most people do not have a studio microphone that can filter out most non-voice sounds. Many only have a headset microphone at best. Push to talk or push to mute used to be the default for that matter, no one likes a person whos microphone is constantly transmitting, and lower quality microphones will always been transmitting.

**Comment by ChthonicLily** [comment]
I will second everything here again. With video and audio calling already added, push to talk settings are going to be a much requested feature by the community as it grows. Yes, technically macros could be set up to mute the mic in the system, but it's so much better for UX to have it in app.

**Comment by jhebert8** [comment]
+1 here for sure. It's indispensable for voice chats for me.

**Comment by Cookie0fPower** [comment]
This needs more attention.

I hope to see PTT (Push-to-Talk) implemented in the near future.

**Comment by sappho114** [comment]
Yeah, really hoping this feature sees attention. I do not want, nor do I need, to hear everyone's mouth sounds. Especially in video games. Fire alarms, barking dogs, people eating - multiple reasons for a PTT option, I hope it's just an oversight and not something difficult to implement. I do not want to use Discord anymore, but I'd prefer it if I could fart without informing the whole squad.

**Comment by jhebert8** [comment]
Is there any movement on this? It is quite literally the only reason I'm not using Stoat for voice chats....

**Comment by Qewser** [comment]
Its 2026, the best chance to get the ppl to your app and this clowns dish it out without push to talk. Best uninstall ever.

**Reply by mihaicm93** [reply]
> Its 2026, the best chance to get the ppl to your app and this clowns dish it out without push to talk. Best uninstall ever.

I get it, it's 2026 and push-to-talk (or at least mute keybinds) is still missing. A lot of people in this thread feel the same way, and it's a totally fair point.

Just wanted to clarify in case you are not aware: the "clowns" you're referring to are all **volunteers** who donate their free time to this project instead of gaming with friends or spending it with loved ones. 
Everything here is free and open-source because of that.

I've already started working on customizable keybinds for mute/deafen over the past week as it's a feature I want. Push-to-talk is on my radar too, but it's a bit more complex to implement .

Thanks for the feedback and for trying the app. **I'd really appreciate it if we could keep things a bit more respectful going forward**.

**Reply by jhebert8** [reply]
That's all I can ask for, confirmation that it's being worked on. Communication is something that often falls through the cracks on open source project, because there's no real incentive in keeping the "customer base" informed and mitigate expectations, etc. It's usually passionate people working on something they want to see, not marketing grad students. It's to be expected.

So yeah, to summarize, that guy was being so disrespectful, and I am really thankful it was acknowledged and will be implemented sometime in the future.

**Reply by mihaicm93** [reply]
Communication on GitHub isn’t always the best, so just a heads-up: we’ve been posting weekly updates from our meet-ups on r/stoatchat and our [main Stoat server](https://stoat.chat/invite/01F7ZSBSFHQ8TA81725KQCSDDP)

**Reply by MrWilza** [reply]
@mihaicm93
The efforts of all of the developers are absolutely appreciated! I think it's mainly that it's this is an essential feature for many people that they've become accustomed to, so without it, they're essentially unable to use the app. For myself too. It's great to know it's being worked on., thank you for letting us know.

**Comment by FinetalPies** [comment]
I can only add 1 upvote to this suggestion but it is surely the absolute #1 priority. Customizable keybinds for push to mute/toggle mute are fully a requirement for me. Others like leave call would be great too.

**Comment by SavageCore** [comment]
Btw, I'd started on PTT support but struggled with syncing status. Core functionality at least worked. Couldn't get my friends to switch to Stoat so abandoned it. https://github.com/SavageCore/for-desktop/commit/d9b03cc587d87af1ebd64bc30383ff23cfff5cb0 https://github.com/SavageCore/for-web/commit/9afc2d5f29048651e0504670660d6c73a1d3a587

Would be good if someone finished it off!

**Comment by Fancy-Raptor** [comment]
Once again bumping this issue of basic functionality that hasn't been finished in _five years_.

**Comment by naelstrof** [comment]
PTT would be nice! It's also preventing me and my friends from switching. Everything else in stoat seemingly worked great.

**Reply by aredon** [reply]
I suspect the ship has sailed. Discord backed off and Stoat didn't get PTT or even signal it was a priority in time. So now a bunch of people that would have swapped will no longer do so.

**Reply by jhebert8** [reply]
It's even more annoying that other alternatives like Fluxer got it in their beta within a week of being asked...

**Reply by FroggyLP756** [reply]
I also would like to switch full-time to Stoat, but without this feature, I just can't.

**Reply by Fancy-Raptor** [reply]
Highly recommend the Silence! app that mutes your audio output in any program. Having to use a separate program for this feature is embarrassing, but it works. I'm on matrix instead of stoat though.

**Comment by FroggyLP756** [comment]
We need those keybind options. I know coding can be hard (I'm also doing it as a hobby)
But five years!
